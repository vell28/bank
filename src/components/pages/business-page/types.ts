export interface IBusinessPageDispatchProps {
  onOpenCompanyClick: () => void;
  onBusinessAccountClick: () => void;
  onAccountingClick: () => void;
  onBusinessSoftwareClick: () => void;
  onInsuranceClick: () => void;
  onPersonalAssistingClick: () => void;
  onHowItWorksClick: () => void;
}

// even we know we never be changing this component to show something
// we won't break the pattern
// tslint:disable-next-line: no-empty-interface
export type IBusinessPageProps = IBusinessPageDispatchProps
