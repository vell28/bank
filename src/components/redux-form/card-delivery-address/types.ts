import { CardProductType } from 'models/operations/card-settings/entities';
import { IAddress } from '../../../models/operations/order-card';

export interface IDeliveryAddress extends IAddress {
  changeDeliveryAddress: boolean;
}

export interface IDeliveryAddressStateProps {
  productType: CardProductType;
  clientAddress: IAddress;
}

export interface IDeliveryAddressDispatchProps {
  onNext: (data: IAddress) => void;
}
