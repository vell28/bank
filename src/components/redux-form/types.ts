export interface IValue {
  value: string | undefined;
}
