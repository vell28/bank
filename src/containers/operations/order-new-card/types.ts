export interface IOrderCardStateProps {
  step: number;
}

export interface IOrderCardDispatchProps {
  orderCardInit: () => void;
  orderCardCleanup: () => void;
}

export interface IOrderCardProps extends IOrderCardStateProps, IOrderCardDispatchProps {}
