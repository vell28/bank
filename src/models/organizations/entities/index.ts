export * from './account';
export * from './balance';
export * from './client';
export * from './organization';
