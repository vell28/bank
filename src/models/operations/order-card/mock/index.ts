import { IServerAddress } from 'models/organizations/redux/actions/entities';
import { IClient } from 'models/organizations/entities';
import { IAddress, ICardOrderOwner } from '../entities';

export const exampleAddress: IAddress = {
  countryCode: 'DE',
  city: 'Cologne',
  postalCode: '50667',
  street: 'Domkloster',
  streetNumber: '4',
  apartmentNumber: 0,
};

export const exampleOrganizationAddress: IAddress = {
  countryCode: 'DE',
  city: 'Cologne',
  postalCode: '50667',
  street: 'Domkloster',
  streetNumber: '4',
  apartmentNumber: 0,
};

export const exampleClientInfo: ICardOrderOwner = {
  embossedName: 'Frank-Walter Steinmeier',
  phoneNumber: '9111111',
};

export const exampleServerAddress: IServerAddress = {
  country: 'Germany',
  province: 'North Rhine-Westphalia',
  city: 'Düsseldorf',
  postalCode: '40479',
  street: 'Gartenstraße',
  streetNumber: '53',
  apartmentNumber: '42',
};

export const exampleClient: IClient = {
  address: exampleOrganizationAddress,
  name: 'Frank-Walter Steinmeier',
  phoneNumber: '9111111',
};
