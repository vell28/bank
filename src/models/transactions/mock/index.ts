import { CurrencyCodes } from 'modules/currencies';
import { ITransaction, ITransactionResponse } from '../entities';

export const transactionData: ITransaction = {
  id: 'PPY184598',
  amount: -10.0,
  fee: 0,
  status: 'COMPLETED',
  paymentType: 'TRANSFER',
  fromNumber: 'MT98PAPY36836000002676370004519',
  fromName: 'Ralf Williams',
  fromBank: 'Papaya',
  toNumber: 'MT97PAPY36836000002676370003470',
  toName: 'Dmitrii Popov',
  toBank: 'Papaya',
  currency: {
    code: CurrencyCodes.EUR,
  },
  clientId: '6449',
  accountId: 'PPY5107',
  description: 'To: Dmitrii Popov Vvg',
  executedAt: '2019-10-11T10:12:00',
  referenceNumber: 'PPY1e1d31e3-6acb-4d60-9f70-b9151221d11',
  paymentToRepeat: {
    type: 'PAYMENT_PHONE',
    fromPhone: '79111111111',
    account: 'PPY5107',
    fromCardPan: null,
    fromCardId: null,
    beneficiaryName: null,
    beneficiaryAccount: null,
    country: null,
    city: null,
    address: null,
    swiftCode: null,
    beneficiaryBank: null,
    intermediaryBank: null,
    intermediarySwift: null,
    intermediaryAccount: null,
    commissionType: null,
    urgency: null,
    iban: null,
    phoneNumber: '79085491121',
    cardNumber: null,
    currencyCode: CurrencyCodes.EUR,
    amount: 10,
    purpose: 'other Vvgh',
    transferDetails: null,
  },
  openapiData: {
    type: 'TO_CONTACT',
    toCard: null,
    toPhone: '79085491121',
    subType: null,
    externalOwnerID: null,
    fromCardPan: null,
  },
  operationType: null,
  fromPanDisplay: null,
};

const second: ITransaction = {
  id: 'PPY184597',
  amount: -1.0,
  fee: 0,
  status: 'COMPLETED',
  paymentType: 'TRANSFER',
  fromNumber: 'MT98PAPY36836000002676370004519',
  fromName: 'Ralf Williams',
  fromBank: 'Papaya',
  toNumber: 'MT52PAPY36836000002676370000303',
  toName: 'Elfur Long',
  toBank: 'Papaya',
  currency: {
    code: CurrencyCodes.EUR,
  },
  clientId: '6449',
  accountId: 'PPY5107',
  description: 'To: Elfur Long 12',
  executedAt: '2019-10-11T10:08:47',
  referenceNumber: 'PPY84f7087c-c511-4928-b299-13ee47c1bd9',
  paymentToRepeat: {
    type: 'PAYMENT_PHONE',
    fromPhone: '79111111111',
    account: 'PPY5107',
    fromCardPan: null,
    fromCardId: null,
    beneficiaryName: null,
    beneficiaryAccount: null,
    country: null,
    city: null,
    address: null,
    swiftCode: null,
    beneficiaryBank: null,
    intermediaryBank: null,
    intermediarySwift: null,
    intermediaryAccount: null,
    commissionType: null,
    urgency: null,
    iban: null,
    phoneNumber: '79032950311',
    cardNumber: null,
    currencyCode: CurrencyCodes.EUR,
    amount: 1,
    purpose: 'other 123',
    transferDetails: null,
  },
  openapiData: {
    type: 'TO_CONTACT',
    toCard: null,
    toPhone: '79032950311',
    subType: null,
    externalOwnerID: null,
    fromCardPan: null,
  },
  operationType: null,
  fromPanDisplay: null,
};

export const transactionList = [transactionData, second];

export const transactionResponse: ITransactionResponse = {
  content: transactionList,
  maxPageNumber: 10,
  pageSize: 20,
  pageNumber: 1,
  length: 0,
};
