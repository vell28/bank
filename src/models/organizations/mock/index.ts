import { CurrencyCodes } from 'modules/currencies';
import {
  IAccount, AccountStatus, AccountTypes, AccountColors
} from '../entities';
import { ICard, CardProduct, Limit } from '../../operations/card-settings/entities';

import { IServerClient } from '../redux/actions/entities';
import { exampleServerAddress } from '../../operations/order-card/mock';

export const cardData: ICard = {
  accountId: 'PPY5107',
  availableBalance: 1897.61,
  balance: 0,
  balanceHold: 0,
  balances: [
    {
      currency: CurrencyCodes.EUR,
      balance: 1030.05,
      balanceHold: 0,
      availableBalance: 1897.61,
    },
  ],
  clientId: '6449',
  currency: { code: CurrencyCodes.EUR },
  expireAt: '2022-05-01T00:00:00',
  id: 'PPY13643',
  limits: [
    {
      type: Limit.DAY,
      title: null,
      maxValue: 500,
      usedValue: 0,
    },
  ],
  number: '5277 60_3 984',
  options: {
    limits: { disable: false },
  },
  owner: { embossedName: 'RALF WILLIAMS' },
  paySystem: 'MASTERCARD',
  plasticStatus: 'PENDING_SHIPMENT',
  productType: CardProduct.MAIN,
  status: AccountStatus.ACTIVE,
};

export const accountsData: IAccount[] = [
  {
    accountName: 'MT98PAPY36836000002676370004519',
    accountType: AccountTypes.PHYSICAL,
    accruedInterest: null,
    availableBalance: 1897.61,
    availableOperations: [],
    balance: 0,
    balanceHold: 0,
    color: AccountColors.DEFAULT,
    balances: [
      {
        currency: CurrencyCodes.EUR,
        balance: 1030.05,
        balanceHold: 0,
        availableBalance: 1897.61,
      },
    ],
    bonusProgramActivate: false,
    clientId: '6449',
    createdAt: '2019-11-24T14:59:55',
    currency: { code: CurrencyCodes.EUR },
    id: 'PPY5107',
    number: 'MT98PAPY36836000002676370004519',
    status: AccountStatus.ACTIVE,
  },
  {
    accountName: 'MT98PAPY36836000002676370004519',
    accountType: AccountTypes.JURIDICAL,
    accruedInterest: null,
    availableBalance: 1897.61,
    availableOperations: [],
    balance: 0,
    balanceHold: 0,
    color: AccountColors.DEFAULT,
    balances: [
      {
        currency: CurrencyCodes.EUR,
        balance: 1030.05,
        balanceHold: 0,
        availableBalance: 1897.61,
      },
    ],
    bonusProgramActivate: false,
    clientId: '6449',
    createdAt: '2019-11-24T14:59:55',
    currency: { code: CurrencyCodes.EUR },
    id: 'PPY6666',
    number: 'MT98PAPY36836000002676370004519',
    status: AccountStatus.ACTIVE,
  },
];

export const clientData: IServerClient = {
  id: '42',
  name: 'Test Client',
  citizenship: 'Germany',
  email: 'test.client@test.com',
  phoneNumber: '222333',
  address: exampleServerAddress,
};
