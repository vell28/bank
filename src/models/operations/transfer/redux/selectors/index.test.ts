import { accountsData } from 'models/organizations/mock';
import { emptyStore } from 'modules/store/emptyStore';
import { IStore } from 'modules/store/types';
import {
  getTransferAmountByValue,
  getCard,
  getPaymentCardData,
  getPaymentSepaData,
  getPaymentSwiftData,
  getPaymentContactData,
} from '.';

const cardData = {
  cardNumber: '123 123',
  beneficiaryName: 'Test',
  expiresDate: '',
  cvv: '000',
};

const phone = '123123123';

const store: IStore = {
  ...emptyStore,
  operations: {
    ...emptyStore.operations,
    transfer: {
      isLoading: false,
      isSuccess: false,
      isError: false,
      data: {
        card: cardData,
        purpose: 'test',
        value: '10',
        phone,
        bank: {},
      },
      errors: [],
    },
  },
  router: {
    action: 'PUSH',
    location: {
      pathname: `/banking/account/${accountsData[0].id}/EUR`,
      search: '',
      hash: '',
      key: '',
      state: '',
    },
  },
};

const sepaData = {
  beneficiaryName: 'beneficiaryName',
  iban: 'MT422PAPY368360000026763123123',
  purpose: 'purpose',
  transferDetails: 'transferDetails',
};

const sepaStore: IStore = {
  ...store,
  operations: {
    ...emptyStore.operations,
    transfer: {
      isLoading: false,
      isSuccess: false,
      isError: false,
      data: {
        card: cardData,
        purpose: 'test',
        value: '20',
        phone,
        bank: sepaData,
      },
      errors: [],
    },
  },
};

const swiftData = {
  beneficiaryName: 'beneficiaryName',
  iban: 'LV21HABA0001408032543',
  country: 'LV',
  city: 'Riga',
  address: 'Riga, Hipokrata, 22',
  swiftCode: 'HABALV22',
  beneficiaryBank: 'SB',
  intermediaryBank: 'intermediaryBank',
  intermediarySwift: 'intermediarySwift',
  intermediaryAccount: 'intermediaryAccount',
  commissionType: 'SENDER',
  urgency: 'NORMAL',
  purpose: 'purpose',
};

const swiftStore: IStore = {
  ...store,
  operations: {
    ...emptyStore.operations,
    transfer: {
      isLoading: false,
      isSuccess: false,
      isError: false,
      data: {
        card: cardData,
        purpose: 'test',
        value: '30',
        phone,
        bank: swiftData,
      },
      errors: [],
    },
  },
};

describe('transfer to card selector tests', () => {
  test('getTransferAmountByValue should return amount by value', () => {
    const amountSelected = getTransferAmountByValue(store);
    expect(amountSelected.sum.value).toEqual('10');
    expect(amountSelected.sum.currency.code).toEqual('EUR');
  });

  test('getCard should return card', () => {
    const card = getCard(store);
    expect(card).toBe(cardData);
    expect(card.cvv).toEqual('000');
  });

  test('getPaymentCardData should return paymentContact of card request', () => {
    const contactData = getPaymentCardData(store);
    expect(contactData.payment_contact.account).toBe(accountsData[0].id);
    expect(contactData.payment_contact.amount.sum.value).toEqual('10');
    expect(contactData).toMatchSnapshot();
  });

  test('getPaymentSepaData should return paymentSepa', () => {
    const bank = getPaymentSepaData(sepaStore);
    expect(bank.payment_sepa.account).toEqual(accountsData[0].id);
    expect(bank.payment_sepa.amount.sum.value).toEqual('20');
    expect(bank.payment_sepa.beneficiaryName).toEqual(sepaData.beneficiaryName);
    expect(bank.payment_sepa.iban).toEqual(sepaData.iban);
    expect(bank.payment_sepa.purpose).toEqual(sepaData.purpose);
    expect(bank.payment_sepa.transferDetails).toEqual(sepaData.transferDetails);
    expect(bank).toMatchSnapshot();
  });

  test('getPaymentSwiftData should return paymentSwift', () => {
    const bank = getPaymentSwiftData(swiftStore);
    expect(bank.payment_swift.account).toEqual(accountsData[0].id);
    expect(bank.payment_swift.amount.sum.value).toEqual('30');
    expect(bank.payment_swift.beneficiaryName).toEqual(swiftData.beneficiaryName);
    expect(bank.payment_swift.beneficiaryAccount).toEqual(swiftData.iban);
    expect(bank.payment_swift.country).toEqual(swiftData.country);
    expect(bank.payment_swift.city).toEqual(swiftData.city);
    expect(bank.payment_swift.address).toEqual(swiftData.address);
    expect(bank.payment_swift.swiftCode).toEqual(swiftData.swiftCode);
    expect(bank.payment_swift.beneficiaryBank).toEqual(swiftData.beneficiaryBank);
    expect(bank.payment_swift.intermediaryBank).toEqual(swiftData.intermediaryBank);
    expect(bank.payment_swift.intermediarySwift).toEqual(swiftData.intermediarySwift);
    expect(bank.payment_swift.intermediaryAccount).toEqual(swiftData.intermediaryAccount);
    expect(bank.payment_swift.commissionType).toEqual(swiftData.commissionType);
    expect(bank.payment_swift.urgency).toEqual(swiftData.urgency);
    expect(bank.payment_swift.purpose).toEqual(swiftData.purpose);
    expect(bank).toMatchSnapshot();
  });

  test('getPaymentContactData should return paymentContact of contact request', () => {
    const contact = getPaymentContactData(store);
    expect(contact.payment_contact.phoneNumber).toEqual(phone);
    expect(contact.payment_contact.amount.sum.value).toEqual('10');
    expect(contact).toMatchSnapshot();
  });
});
