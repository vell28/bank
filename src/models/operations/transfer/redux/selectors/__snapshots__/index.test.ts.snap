// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transfer to card selector tests getPaymentCardData should return paymentContact of card request 1`] = `
Object {
  "payment_contact": Object {
    "account": "PPY5107",
    "amount": Object {
      "sum": Object {
        "currency": Object {
          "code": "EUR",
        },
        "value": "10",
      },
    },
    "beneficiaryName": "Test",
    "cardNumber": "123123",
    "fromCardId": "",
    "purpose": "test",
  },
}
`;

exports[`transfer to card selector tests getPaymentContactData should return paymentContact of contact request 1`] = `
Object {
  "payment_contact": Object {
    "account": "PPY5107",
    "amount": Object {
      "sum": Object {
        "currency": Object {
          "code": "EUR",
        },
        "value": "10",
      },
    },
    "phoneNumber": "123123123",
    "purpose": "test",
  },
}
`;

exports[`transfer to card selector tests getPaymentSepaData should return paymentSepa 1`] = `
Object {
  "payment_sepa": Object {
    "account": "PPY5107",
    "amount": Object {
      "sum": Object {
        "currency": Object {
          "code": "EUR",
        },
        "value": "20",
      },
    },
    "beneficiaryName": "beneficiaryName",
    "iban": "MT422PAPY368360000026763123123",
    "purpose": "purpose",
    "transferDetails": "transferDetails",
  },
}
`;

exports[`transfer to card selector tests getPaymentSwiftData should return paymentSwift 1`] = `
Object {
  "payment_swift": Object {
    "account": "PPY5107",
    "address": "Riga, Hipokrata, 22",
    "amount": Object {
      "sum": Object {
        "currency": Object {
          "code": "EUR",
        },
        "value": "30",
      },
    },
    "beneficiaryAccount": "LV21HABA0001408032543",
    "beneficiaryBank": "SB",
    "beneficiaryName": "beneficiaryName",
    "city": "Riga",
    "commissionType": "SENDER",
    "country": "LV",
    "intermediaryAccount": "intermediaryAccount",
    "intermediaryBank": "intermediaryBank",
    "intermediarySwift": "intermediarySwift",
    "purpose": "purpose",
    "swiftCode": "HABALV22",
    "urgency": "NORMAL",
  },
}
`;
